sdk/web-pubsub/web-pubsub/src/hubClient.ts

```diff
@@ -0,0 +1,659 @@
+   */
+  constructor(connectionString: string, hubName: string, options?: HubAdminClientOptions);
+
+  /**
+   * Creates an instance of a HubClient for sending messages and managing groups, connections, and users.
+   *
+   * Example usage:
+   * ```ts
+   * import { HubClient, AzureKeyCredential } from "@azure/web-pubsub-management";
+   * const cred = new AzureKeyCredential("<your web pubsub api key>");
+   * const endpoint = "https://xxxx.webpubsubdev.azure.com"
+   * const client = new HubClient(endpoint, cred, 'chat');
+   * ```
+   *
+   * @param endpoint The endpoint to connect to
+   * @param credential An AzureKeyCredential holding your service key
+   * @param hubName The name of the hub to connect to.
+   * @param options Options to configure the http pipeline
+   */
+  constructor(
+    endpoint: string,
+    credential: AzureKeyCredential,
+    hubName: string,
+    options?: HubAdminClientOptions
+  );
+  constructor(
+    epOrCs: string,
+    credsOrHubName?: AzureKeyCredential | string,
+    hubNameOrOpts?: string | HubAdminClientOptions,
+    opts?: HubAdminClientOptions
+  ) {
+    // unpack constructor arguments
+    if (typeof credsOrHubName === "object" && "key" in credsOrHubName) {
+      this.endpoint = epOrCs;
+      this.credential = credsOrHubName;
+      this.hubName = hubNameOrOpts as string;
+      this.clientOptions = opts;
+    } else {
+      const parsedCs = parseConnectionString(epOrCs);
+      this.endpoint = parsedCs.endpoint;
+      this.credential = parsedCs.credential;
+      this.hubName = credsOrHubName as string;
+      this.clientOptions = hubNameOrOpts as PipelineOptions;
+    }
+
+    const internalPipelineOptions: InternalPipelineOptions = {
+      ...this.clientOptions,
+      ...{
+        loggingOptions: {
+          logger: logger.info
+        }
+      }
+    };
+
+    const pipeline = createPipelineFromOptions(
+      internalPipelineOptions,
+      webPubSubAzureKeyCredentialPolicyFactory(this.credential)
+    );
+
+    this.client = new GeneratedClient(this.endpoint, pipeline);
+  }
+
+  /**
+   * Auth the client connection with userId and custom claims if any
+   * @param options The options that the client has
+   */
+  public async getAuthenticationToken(
+    options?: GetAuthenticationTokenOptions
+  ): Promise<GetAuthenticationTokenResponse> {
+    const endpoint = this.endpoint.endsWith("/") ? this.endpoint : this.endpoint + "/";
+    const key = this.credential.key;
+    const hub = this.hubName;
+    var clientEndpoint = endpoint.replace(/(http)(s?:\/\/)/gi, "ws$2");
+    var clientUrl = `${clientEndpoint}client/hubs/${hub}`;
+    const audience = `${endpoint}client/hubs/${hub}`;
+    var payload = options?.claims ?? {};
+    var signOptions: jwt.SignOptions = {
+      audience: audience,
+      expiresIn: "1h",
+      algorithm: "HS256"
+    };
+    if (options?.userId) {
+      signOptions.subject = options?.userId;
+    }
+    const token = jwt.sign(payload, key, signOptions);
```

Comment:
I think `jwt.sign` has an async version via callbacks. Worth wrapping in a promisified version so we don't need to do sync operations?

Reply:
I'm not sure. Do you think so?

Reply:
I looked at the dependency chain, and no, I don't think it's worth it. The underlying crpyto operations are being called in a sync way so I don't think we'd see real benefit.